src/tests/ports/nfts-rentalsNFTComponent.spec.ts

```diff
@@ -0,0 +1,357 @@
+import { Contract, NFTFilters } from '@dcl/schemas'
+import {
+  createRentalsNFTComponent,
+  getLandAndEstateContractAddresses,
+  MAX_SUBGRAPH_QUERY_IN_ELEMENTS,
+  rentalNFTComponentShouldFetch,
+} from '../../ports/nfts/rentalsNFTComponent'
+import { INFTsComponent } from '../../ports/nfts/types'
+import { IRentalsComponent, RentalAsset } from '../../ports/rentals/types'
+
+let mockMarketplaceNFTsComponent: INFTsComponent
+let mockRentalsComponent: IRentalsComponent
+let mockContractAddresses: { land: string; estate: string }
+
+let rentalsNFTComponent: INFTsComponent
+
+beforeEach(() => {
+  mockMarketplaceNFTsComponent = {
+    count: jest.fn(),
+    fetch: jest.fn(),
+    fetchByTokenIds: jest.fn(),
+    fetchOne: jest.fn(),
+  }
+
+  mockRentalsComponent = {
+    getRentalAssets: jest.fn().mockResolvedValue([{}]),
+    getRentalsListings: jest.fn(),
+    getRentalsListingsOfNFTs: jest.fn(),
+  }
+
+  mockContractAddresses = { land: 'land', estate: 'estate' }
+
+  rentalsNFTComponent = createRentalsNFTComponent({
+    contractAddresses: mockContractAddresses,
+    marketplaceNFTsComponent: mockMarketplaceNFTsComponent,
+    rentalsComponent: mockRentalsComponent,
+  })
+})
+
+describe('when fetching nfts', () => {
+  describe('when calling getRentalAssets from the IRentalsComponent', () => {
+    it('should be called with the provided filters owner in the lessors array', async () => {
+      await rentalsNFTComponent.fetch({ owner: 'owner' })
```

Comment:
What if you move the fetch execution into a beforeEach to avoid repeating the call?

Reply:
On all tests?